src/Compilers/CSharp/Portable/Symbols/Extensions/SourceExtensionImplementationMethodSymbol.cs

```diff
@@ -175,6 +206,27 @@ public override int Ordinal
                     return ordinal + 1;
                 }
             }
+
+            internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+            {
+                // Copy AllowNull/DisallowNull from the property onto the implementation setter's `value` parameter
+                if (this._underlyingParameter is SynthesizedAccessorValueParameterSymbol
+                    && this._underlyingParameter.ContainingSymbol is SourcePropertyAccessorSymbol { MethodKind: MethodKind.PropertySet, AssociatedSymbol: SourcePropertySymbolBase extensionProperty })
```

Comment:
Similar suggestion as for ```AddSynthesizedReturnTypeAttributes```. I.e. extract shared helper from ```SynthesizedAccessorValueParameterSymbol.AddSynthesizedAttributes``` and call it from here on _underlyingParameter instead of trying to duplicate the logic inside this `if` statement.